home-application/src/main/java/com/softserveinc/ita/homeproject/application/config/EnumConverterProvider.java

```diff
@@ -0,0 +1,47 @@
+package com.softserveinc.ita.homeproject.application.config;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.ext.ParamConverter;
+import javax.ws.rs.ext.ParamConverterProvider;
+import javax.ws.rs.ext.Provider;
+
+@SuppressWarnings("unchecked")
+@Provider
+public class EnumConverterProvider implements ParamConverterProvider {
+
+    @Override
+    public <T> ParamConverter<T> getConverter(Class<T> rawType, Type genericType, Annotation[] annotations) {
+        if (!rawType.isEnum()) {
+            return null;
+        }
+
+        final Class<Enum<?>> type = (Class<Enum<?>>) rawType;
+        final Enum<?>[] constants = type.getEnumConstants();
+
+        return new ParamConverter<T>() {
+            @Override
+            public T fromString(String value) {
+                if (value == null || value.isEmpty()) {
+                    return null;
+                }
+
+                for (Enum<?> constant : constants) {
+                    if (constant.toString().equals(value)) {
+                        return (T) constant;
+                    }
+                }
+
+                throw new BadRequestException(String.format("The parameter type must be one of the values: %s",
+                    Arrays.toString(Arrays.stream(constants).toArray())));
```

Comment:
This peace of code looks strange:
```Arrays.toString(Arrays.stream(constants).toArray())```
Can we just do: `Arrays.toString(constants)`? 
Is the result is csv?